### DISCUSSION TITLE
IoT core IPC - AWS SDK - Publish to IoT Core Sample code (no network connection)

### DISCUSSION THREAD

**Discussion by chandar1** [aws/aws-iot-device-sdk-cpp-v2#768] [Q&A]
I'm using the sample CPP code given AWS SDK Documentation (below) - to connected to Nucleus and do IPC PublishToIoTCore.

When network connection don't exist or IoT core connection don't exist, then this line `auto response = responseFuture.get();` blocks thread indefinitely. 

I would need to be able to publish a number of messages to IoT core and only have limited number of threads. In the no-network-connection scenario how could I ensure, thread doesn't block, or check for IoT core connected or not using aws-sdk and then initiate publish messages to iot core.

Any help greatly appreciated.
```
part of sample code
.....................
    auto operation = ipcClient.NewPublishToIoTCore();
    auto activate = operation->Activate(request, nullptr);
    activate.wait();

    auto responseFuture = operation->GetResult();
    if (responseFuture.wait_for(std::chrono::seconds(timeout)) == std::future_status::timeout) {
        std::cerr << "Operation timed out while waiting for response from Greengrass Core." << std::endl;
        exit(-1);
    }

    auto response = responseFuture.get();
    if (!response) {
        // Handle error.
        auto errorType = response.GetResultType();
        if (errorType == OPERATION_ERROR) {
            auto *error = response.GetOperationError();
            (void)error;
            // Handle operation error.
        } else {
            // Handle RPC error.
        }
    }
...................................
```

**Comment by sfod** [comment]
Are you sure your code blocks on the `auto response = responseFuture.get();` line? I'm asking because above this line you call `responseFuture.wait_for(...)`, so `responseFuture` should have some result.

On the other hand, `activate.wait();` doesn't provide timeout, so it can in theory blocks for a long time.
